backend/server/mailing.go

```diff
@@ -2,91 +2,239 @@ package main
 
 import (
 	"net/http"
+	"time"
 
 	"github.com/labstack/echo/v4"
+	"github.com/mailjet/mailjet-apiv3-go"
 )
 
-// Message - struct to contain email message data
-type Message struct {
+// Enquiry - struct to contain email enquiry data
+type Enquiry struct {
 	Name  string `validate:"required"`
 	Email string `validate:"required,email"`
 	Body  string `validate:"required"`
 }
 
 // Feedback - struct to contain feedback message data
-// name is not required, email must be valid and body is required.
+// name is not required, email must be valid (or empty) and body is required.
 type Feedback struct {
 	Name  string
-	Email string `validate:"email"`
+	Email string `validate:"omitempty,email"`
 	Body  string `validate:"required"`
 }
 
-// FeedbackNoEmail - struct to contain feedback message when there is no email given
-// name, email are not required, body is.
-type FeedbackNoEmail struct {
-	Name  string
-	Email string
-	Body  string `validate:"required"`
+// Message bundles
+var feedbackBundle []Feedback
+var infoBundle []Enquiry
+var infoEmail string = "info@csesoc.org.au"
+var sponsorshipBundle []Enquiry
+var sponsorshipEmail string = "sponsorship@csesoc.org.au"
+
+// Mailjet session variables
+var publicKey string = "8afb96baef07230483a2a5ceca97d55d"
+var secretKey string = "424ad90f25487e6be369a1cbb2a34694"
+var mailjetClient *mailjet.Client
+
+// InitMailClient initialises a session with the Mailjet API and stores it in a global variable
+func InitMailClient() {
+	mailjetClient = mailjet.NewMailjetClient(publicKey, secretKey)
+
+	// Start mailing timers
+	go enquiryMailingTimer()
+	go feedbackMailingTimer()
 }
 
+///////////
+// HANDLERS
+///////////
+
 // HandleEnquiry by forwarding emails to relevant inboxes
 func HandleEnquiry(targetEmail string) echo.HandlerFunc {
 	return func(c echo.Context) error {
-		message := Message{
+		// Extract fields from form
+		enquiry := Enquiry{
 			Name:  c.FormValue("name"),
 			Email: c.FormValue("email"),
 			Body:  c.FormValue("body"),
 		}
 
-		if err := c.Validate(message); err != nil {
+		// Validate struct
+		if err := c.Validate(enquiry); err != nil {
 			return c.JSON(http.StatusBadRequest, H{
 				"error": err,
 			})
 		}
 
-		// TODO: Compose and send email
-		return c.JSON(http.StatusOK, H{})
+		// Add to bundle
+		switch targetEmail {
+		case sponsorshipEmail:
+			sponsorshipBundle = append(sponsorshipBundle, enquiry)
+		case infoEmail:
+			infoBundle = append(infoBundle, enquiry)
+		}
+
```

Comment:
In a default case maybe we should return an error code

Reply:
In the case where the passed email isn't neither of the cases we can assume that it's a developer error since this function is called by the backend. Adding a default case will return an error to the client, instead of the server.

I was thinking on defining these emails as a const on a global variable file (remember utils.go you mentioned on Friday) so that these emails are only defined once.